cmd/xl-v1-multipart.go

```diff
@@ -833,6 +833,7 @@ func (xl xlObjects) listObjectParts(bucket, object, uploadID string, partNumberM
 	result.Object = object
 	result.UploadID = uploadID
 	result.MaxParts = maxParts
+	result.PartNumberMarker = partNumberMarker
```

Comment:
should this be fixed outside this PR?

Reply:
unit tests were failing which required this fix. The refactor exposed this bug.

Reply:
Can you add in a separate PR ? This should be reproducible outside of this PR as well right ?

Reply:
Actually, making the changes I fixed this bug and this broke the tests (as tests were itself incorrect), so the tests needed a fix as well. I can do this in separate PR and retain the bug in the code and tests in this PR. Probably the effort is not worth it at this point since it's all fixed.

Reply:
Okay then it's fine.